numpy/core/include/numpy/npy_cpu.h

```diff
@@ -115,4 +116,83 @@
 #define NPY_CPU_HAVE_UNALIGNED_ACCESS 0
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum
+{
+    NPY_OPT_NONE = 0,
```

Comment:
do we want to make cpu detection part of our api?
could be useful for other projects

though the name `NPY_OPT` is not really clear enough, maybe `NPY_CPU_FEATURE`

Reply:
> do we want to make cpu detection part of our api?
 could be useful for other projects

yes, not even cpu detection part but also `universal intrinsics`

> though the name NPY_OPT is not really clear enough, maybe NPY_CPU_FEATURE

I agree I like `NPY_CPU_{FEATURE}` more but the prefix `NPY_CPU_` already taken by CPU names, so I'm going to use `NPY_CPU_FEATURE_` as a prefix instead.